src/js/edu-benefits/components/ReviewCollapsiblePanel.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import Scroll from 'react-scroll';
+import _ from 'lodash';
+
+import * as validations from '../utils/validations';
+import { pages } from '../routes';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+/**
+ * A component for the review page to validate information is correct.
+ *
+ * Required props
+ */
+
+export default class ReviewCollapsiblePanel extends React.Component {
+  constructor() {
+    super();
+    this.handleSave = this.handleSave.bind(this);
+    this.handleNext = this.handleNext.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.scrollToTop = this.scrollToTop.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId();
+  }
+
+  scrollToTop() {
+    scroller.scrollTo('topScrollReviewPanel', {
+      duration: 500,
+      delay: 2,
+      smooth: true,
+    });
+  }
+
+  handleSave() {
+    const currentPath = this.props.updatePath;
+    const formData = this.props.data;
+    const pageFields = this.props.uiData.pages[currentPath].fields;
+
+    this.props.onFieldsInitialized(pageFields);
+    if (validations.isValidPage(currentPath, formData)) {
+      this.props.onUpdateSaveStatus(currentPath);
+    }
+    this.scrollToTop();
+  }
+
+  handleNext() {
+    const currentPath = this.props.updatePath;
+    this.props.onUpdateVerifiedStatus(currentPath, true);
+    // TODO: find a better solution for this or a different implementation.
+    setTimeout(() => this.scrollToTop(), 100);  // eslint-disable-line scanjs-rules/call_setTimeout
+  }
+
+  handleEdit() {
+    const currentPath = this.props.updatePath;
+    this.props.onUpdateEditStatus(currentPath);
+    this.props.onUpdateVerifiedStatus(currentPath, false);
+    this.scrollToTop();
+  }
+
+  allPreviousPagesVerified(pageState, data, currentPath) {
+    const filteredPages = pages.filter(page => {
+      return page.depends === undefined || _.matches(page.depends)(data);
+    });
```

Comment:
This function might be best as something in `utils`, since we call it in a few places now.  Maybe something like `getActivePages`.